handlers/chat.go

```diff
@@ -720,6 +720,40 @@ func (ch *ChatHandler) ProcessChatResponse(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
+	existingMessages, err := ch.db.GetChatMessagesForChatID(request.Value.ChatID)
+	if err != nil {
```

Comment:
I don't think this logic is correct @aliraza556. 
When we send a chat request to Stakwork we log the request in DB and include the Message ID of that chat request as message ID to stakwork. 
The message ID that we receive back from Stakwork is the reference to the message we are responding to. 
So if I read this logic correctly this would always resolve in messages not being recorded as the Message ID would always already exist.

Reply:
Should be a ref to line 733